### PR TITLE
bump k8s to 1.25 and go to 1.19

### PR DESCRIPTION
**Pull request by laxmikantbpandhare**
Ready for review.

### CODE REVIEW

**Comment by everettraven** [comment]
@laxmikantbpandhare It seems the `verify` check is failing when running a `git diff` from the generated code. I think you may be able to fix this by running `make generate` and then committing the changes.

**Comment by laxmikantbpandhare** [comment]
I tried that command already but did not get anything for commit.

```
laxmikantbhaskarpandhare@lpandhar-mac api % make generate
/Users/laxmikantbhaskarpandhare/go/src/github.com/operator-frameowrk/api/bin/controller-gen object:headerFile=./hack/boilerplate.go.txt paths=./...
laxmikantbhaskarpandhare@lpandhar-mac api % git status
On branch update-1.25
nothing to commit, working tree clean
laxmikantbhaskarpandhare@lpandhar-mac api % 
```

**Comment by everettraven** [comment]
@laxmikantbpandhare it looks like the verify check is still failing based on some formatting errors. I think doing a `make format` may resolve these.

**Comment by laxmikantbpandhare** [comment]
Ah! yeah it was still failing. I made changes to it.

**Review by everettraven — APPROVED** [review]
/lgtm

**Review by jmrodri — APPROVED** [review]
/lgtm

**Comment by awgreene** [comment]
Seems like the `go-apidiff` test is failing due to [this change](https://github.com/kubernetes/apimachinery/commit/430b920312ca0fa10eca95967764ff08f34083a3#diff-fcf220c1f297335eaed6206c361a29bdce987823d449f8bd83ae753000739a9fL133) and can be ignored.

**Comment by awgreene** [comment]
If there are no major objections, I would like to see #261 merge first, as it isn't clear how much work it will be to update OLM to use the new k8s and go versions.

/hold

**Comment by laxmikantbpandhare** [comment]
> If there are no major objections, I would like to see #261 merge first, as it isn't clear how much work it will be to update OLM to use the new k8s and go versions.
> 
> /hold

@awgreene - this will impact SDK effort https://github.com/operator-framework/operator-sdk/issues/5937. @jmrodri @theishshah  - Can we wait for this?

**Comment by anik120** [comment]
Fyi #261 is just waiting on an `approve` label, which should be there soon. So we won't have to wait long for that.

**Comment by anik120** [comment]
#261 has merged. Fyi I'm cutting a new release so that we can use that in OLM. 

/hold cancel

File: pkg/validation/internal/annotations.go [hunk 1]
```diff
@@ -30,13 +30,13 @@ which are known to be case sensitive. It also checks to see if the olm.propertie
 annotation is defined in order to add a warning if present. This function can be
 used anywhere annotations need to be checked for case sensitivity.
 
-Arguments
+# Arguments
```
Comment (awgreene) [inline pkg/validation/internal/annotations.go:33]: Could you explain these changes?

File: pkg/validation/internal/operatorhub.go [hunk 2]
```diff
@@ -71,15 +71,15 @@ import (
...
-// 	```
+//		```json
+//		{
+//			"categories":[
+//	     "Cloud Pak",
+//	     "Registry",
+//	     "MyCoolThing",
+//	 	 ]
+//		}
+//		```
```
Comment (awgreene) [inline pkg/validation/internal/operatorhub.go:82]: Could you explain these changes?

File: pkg/validation/internal/annotations.go (hunk 1 above)
Comment (everettraven) [reply pkg/validation/internal/annotations.go:33]: This has to do with gofmt updates related to bumping the Go version to Go 1.19.

If you take a look at the Headings section of this documentation https://tip.golang.org/doc/comment you can see that as of Go 1.19 the new heading format is similar to Markdown and uses the # prefix:

>    The # syntax was added in Go 1.19. Before Go 1.19, headings were identified implicitly by single-line paragraphs satisfying certain conditions, most notably the lack of any terminating punctuation.
>
>    Gofmt reformats [lines treated as implicit headings](https://github.com/golang/proposal/blob/master/design/51082-godocfmt.md#headings) by earlier versions of Go to use # headings instead. If the reformatting is not appropriate—that is, if the line was not meant to be a heading—the easiest way to make it a paragraph is to introduce terminating punctuation such as a period or colon, or to break it into two lines.

File: pkg/validation/internal/operatorhub.go (hunk 2 above)
Comment (everettraven) [reply pkg/validation/internal/operatorhub.go:82]: I believe this also has to do with Go 1.19 formatting changes. All information on the changes for Go 1.19 doc comment changes should be able to be found here: https://tip.golang.org/doc/comment

Comment (laxmikantbpandhare) [reply pkg/validation/internal/operatorhub.go:82]: thank you @everettraven for your explanation.

Comment (laxmikantbpandhare) [reply pkg/validation/internal/operatorhub.go:82]: These changes are more due to 1.19 formatting changes.

**Pull request merged**